### PR TITLE
Allow to use any background job library to handle async jobs

### PR DESCRIPTION
**Pull request by mpraglowski**
Breaking change: signature of dispatcher's proxy strategy is changed

### CODE REVIEW

File: rails_event_store/spec/async_dispatcher_spec.rb [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      end
+      def call(event)
+        @@received = event
+      end
+    end
+    class NotCallableHandler
+    end
+
+    class MyAsyncHandler
+      @@received = nil
```
Comment (pawelpacana) [inline rails_event_store/spec/async_dispatcher_spec.rb]: I'm afraid this and other class variables will make this test brittle

File: rails_event_store/lib/rails_event_store/async_dispatcher.rb [hunk 2]
```diff
@@ -1,27 +1,25 @@
 module RailsEventStore
```
Comment (pawelpacana) [inline rails_event_store/lib/rails_event_store/async_dispatcher.rb]: `ActiveJobDispatcher` is `RailsEventStore::`, however `AsyncDispatcher` can be moved to `RubyEventStore::` and be used in non-rails scenarios.

Comment (mpraglowski) [reply rails_event_store/lib/rails_event_store/async_dispatcher.rb]: We still use there `AsyncProxyStrategy::AfterCommit` which uses `ActiveRecord` but maybe this should stay in `RailsEventStore::` and only `AsyncProxyStrategy::Inline` should moved with `AsyncDispatcher` to `RubyEventStore::`

Comment (pawelpacana) [reply rails_event_store/lib/rails_event_store/async_dispatcher.rb]: :+1:

**Comment by paneq** [comment]
Looks good (and I am happy it was merged) except for one thing. `async_call` is a suboptimal name. Let's figure out something better? 

How about 

* `scheduler`, 
*`queuing`, 
* `enqueuing`, 
* `enqueue`,
* `enqueuer`, 
* `adapter`

?

**Comment by paneq** [comment]
@mpraglowski do you plan updating the documentation in a separate PR?

**Comment by pawelpacana** [comment]
> async_call is a suboptimal name

`scheduler` and `executor` were on my list

**Comment by pawelpacana** [comment]
btw. docs can live in master via toggle

**Pull request merged**